src/stan-users-guide/gaussian-processes.Rmd

```diff
@@ -1086,3 +1151,98 @@ generated quantities {
+
+### The Squared Exponential
+
+A common choice for a covariance function is the Squared Exponential covariance function. This is also known as the Radial Basis Function or the Gaussian kernel, due to the fact that its function resembles a gaussian. The functional form is as follows
+
+$$
+K(x, x' \mid \sigma, l) = \sigma^2 \exp\left(-\dfrac{||x - x'||_2^2}{2 l^2}\right).
+$$
+for $\sigma,l \geq 0$, where $||\cdot||_2^2$ is the squared l-2 norm, and $x$ and $x'$ are observations of design matrix $X$. The $K_{i,j}$ will then be the covariance computed between rows $i$ and $j$ of the design matrix. Here's what draws from the squared exponential kernel look like, before and after being conditioned on data, with fixed hyper priors for the magntitude parameter $\sigma^2$ and length-scale parameter $l$:
+
+//////////////GRAPHIC
+
+We can sample from a multivariate gaussian distribution in Stan by first instantiating a covariance matrix, for example for `vector[D] x[N]`, and parameters `real<lower=0> sigma` and `real<lower=0> length_scale` as
+
+```
+matrix[N, N] K = gp_exp_quad_cov(x, sigma, l)
+```
+
+And then we add noise and sample from a multivariate gaussian distribution using cholesky decomposition:
+
+```
+
+transformed parameters {
+  vector[N] f;
+  matrix[N, N] L_K;
+  K = add_diag(K, epsilon);
+  L_K = choleskey_decompose(K);
+  f = L_K * eta;
+}
+...
+model {
+  eta ~ normal(0, 1);
+}
+```
+Where since $\eta \sim normal(0, 1)$, $f$ is distributed multivariate normal.
+
+We can implement a gaussian process regression model with a Squared Exponential kernel, along with the posterior predictive for the regression model as follows:
+
+
+### The Dot Product Covariance Function
+
+The dot product kernel, or linear kernel, is analogous to a Bayesian linear regression with a normal(0, 1) prior on regression coefficients. This can be combined with other covariance functions to interpolate linear trends in data, but that will be discussed in a later chapter. The analytic form is simply an inner product between observations $x$ and $x'$, with bias parameter $\sigma_0 > 0$ and inner product $(\cdot, \cdot)$:
+
+$$
+K(x, x' \mid \sigma, l) = \sigma_0^2 + (x', x)
+$$
+
+We can define a dot product covariance function almost identicle to defining the squared exponential covariance function, instead as `gp_dot_prod_cov` with a constant parameter $\sigma_0$, for training data `matrix[D] x[N]`, an $N\times D$ matrix as
+
+```
+matrix[N, N] K = gp_dot_prod_cov(x, sigma0)
+```
+
+and for the cross covariance matrix with $N_{pred} \times D$ `matrix[D] x[N_pred] x_pred` as:
+
+```
+matrix[N, N_pred] K gp_dot_prod_cov(x_pred, x, sigma0)
+
+```
+
+Fixing the bias parameter, we can generate samples from the linear kernel (left) and the linear kernel fitting a linear function:
```

Comment:
"we can generate samples from the linear kernel (left) and the linear kernel fitting a linear function:"
-> we can generate samples from a GP with the linear covariance function (left) and a GP with linear covariance function fitting ???: (I don't understand the end and graphic is missing).